.github/workflows/maven.yaml

```diff
@@ -104,40 +107,80 @@ jobs:
           distribution: 'temurin'
           java-version: '17'
 
+      - name: set branch name env variable
```

Comment:
we repeat these steps, "set env variable", "checkout", "download cache". Unfortunately the closest thing to make these steps "workflow-wide" accessible is called "composite actions", but what we want is not supported. So, we will live with repeating the same thing 3 times.